packages/core/src/services/chains/ProviderProcessor/index.ts

```diff
@@ -13,9 +20,21 @@ export async function processResponse({
   aiResult: Awaited<AIReturn<StreamType>>
   documentLogUuid?: string
 }): Promise<ChainStepResponse<StreamType>> {
-  if (aiResult.type === 'text') {
-    return processStreamText({ aiResult, documentLogUuid })
-  }
+  const isObject = aiResult.type === 'object'
+
+  const text = await aiResult.text
 
-  return processStreamObject({ aiResult, documentLogUuid })
```

Comment:
Before AI could return from `streamObject` or `streamText`. And after streaming was finished we format the response accordingly. Now these 2 methods are gone and all is done in this file alway returning `text` and `object?: object | undefined` dependending if the user pass `schema` or not